tests/functional/conftest.py

```diff
@@ -359,18 +351,14 @@ def _merge_request(*, source_branch: str):
             time.sleep(0.5)
         assert mr.merge_status != "checking"
 
-        to_delete.append((mr.iid, source_branch))
+        to_delete.append((mr, mr_branch))
         return mr
 
     yield _merge_request
 
-    for mr_iid, source_branch in to_delete:
-        project.mergerequests.delete(mr_iid)
-        try:
-            project.branches.delete(source_branch)
-        except gitlab.exceptions.GitlabDeleteError:
-            # Ignore if branch was already deleted
-            pass
+    for mr, mr_branch in to_delete:
+        helpers.safe_delete(object=mr)
+        helpers.safe_delete(object=mr_branch)
```

Comment:
I think we don't need a list of tuples even - the order of deletion shouldn't matter. I guess we could just have:

```python
        to_delete += mr, mr_branch
        return mr

    yield _merge_request

    for resource in to_delete:
        helpers.safe_delete(object=resource)
```

Reply:
I think they can still be deleted after closing but looks good like this to me!